### PR TITLE
feat: add Linear integration

### PR DESCRIPTION
**Pull request by burntcarrot**
* add a basic HTTP client for Linear

Signed-off-by: burntcarrot <aadhav.n1@gmail.com>

### CODE REVIEW

File: integrations/linear/linear.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var parsed map[string]interface{}
+	json.Unmarshal([]byte(string(respBody)), &parsed)
+
+	// set issue ID
+	issue.ID = parsed["data"].(map[string]interface{})["issueCreate"].(map[string]interface{})["issue"].(map[string]interface{})["id"].(string)
```
Comment (burntcarrot) [inline integrations/linear/linear.go:109]: I know this looks messy. Do you have any suggestions on how this can be cleaned up a bit?

Comment (burntcarrot) [reply integrations/linear/linear.go:109]: One way is to declare a Result struct:

```go
// Result represents the response structure.
type Result struct {
	Data struct {
		IssueCreate struct {
			Issue struct {
				ID    string `json:"id"`
				Title string `json:"title,omitempty"`
			} `json:"issue,omitempty"`
		} `json:"issueCreate,omitempty"`
	} `json:"data,omitempty"`
}
```

Then unmarshal the response body:

```go
        parsed := Result{}

	err = json.Unmarshal([]byte(string(respBody)), &parsed)
	if err != nil {
		return "", err
	}

	// set issue ID
	issue.ID = parsed.Data.IssueCreate.Issue.ID

```

Comment (souvik-deepsource) [reply integrations/linear/linear.go:109]: Yes, declaring a `result` struct makes it more readable.

File: integrations/linear/linear.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+	if resp.StatusCode == 200 {
```
Comment (souvik-deepsource) [inline integrations/linear/linear.go:111]: Since you're creating a resource (creating a Linear issue), won't the status code be 201?

Comment (burntcarrot) [reply integrations/linear/linear.go:111]: Technically it should be `201`, but for some reason the Linear API tends to return `200 OK`.

**Review by souvik-deepsource — APPROVED** [review]
Keep it rolling!

**Pull request merged**